### ISSUE TITLE
version 1.0.6

### ISSUE THREAD

**Issue by johntom** [continuous-software/node-authorize-net#8]
Hi,
Things are moving along very well with this version but I've hit a snag and hoping you could help. On a positive note the nodemachine I'm writing provides great documentation and will be easy for others to follow when complete(see http://node-machine.org/). The code fails on the sendTransactionRequest failing on throw new GatewayError(). Below is error (I'm hiding id and key). 
Thanks,
John
```
this is a console log of service
 service  { endpoint: 'https://api.authorize.net/xml/v1/request.api',
  API_LOGIN_ID: 'xxxx4Tk9K',
  TRANSACTION_KEY: 'xxxxN24Hqm7C5V98' }
 body  { transactionType: 'authCaptureTransaction',
  amount: 5,
  payment:
   { creditCard:
      { cardNumber: '5424000000000015',
        expirationDate: '1220',
        cardCode: '999' } },
  billTo:
   { firstName: 'john',
     lastName: 'Thomas',
     city: 'NY',
     state: '10011',
     zip: '10011',
     country: 'USA' },
  shipTo:
   { firstName: 'john',
     lastName: 'Thomas',
     city: 'New York',
     state: 'NY',
     zip: '10011',
     country: 'USA' } }
Unhandled rejection TypeError: Cannot read property '0' of undefined
    at C:\A\machinepack-authorizenet\node_modules\authorize-net\lib\AuthorizeNetGateway.js:113:94
    at C:\A\machinepack-authorizenet\node_modules\authorize-net\lib\AuthorizeNetGateway.js:86:16
    at tryCatcher (C:\A\machinepack-authorizenet\node_modules\authorize-net\node_modules\bluebird\js\main\util.js:26:23)

    at Promise._settlePromiseFromHandler (C:\A\machinepack-authorizenet\node_modules\authorize-net\node_modules\bluebird
\js\main\promise.js:507:31)
    at Promise._settlePromiseAt (C:\A\machinepack-authorizenet\node_modules\authorize-net\node_modules\bluebird\js\main\
promise.js:581:18)
    at Promise._settlePromiseAtPostResolution (C:\A\machinepack-authorizenet\node_modules\authorize-net\node_modules\blu
ebird\js\main\promise.js:245:10)
    at Async._drainQueue (C:\A\machinepack-authorizenet\node_modules\authorize-net\node_modules\bluebird\js\main\async.j
s:128:12)
    at Async._drainQueues (C:\A\machinepack-authorizenet\node_modules\authorize-net\node_modules\bluebird\js\main\async.
js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (C:\A\machinepack-authorizenet\node_modules\authorize-net\node_modu
les\bluebird\js\main\async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:358:17)
```

**Comment by lorenzofox3** [comment]
seems an error while [parsing the response](https://github.com/continuous-software/node-authorize-net/blob/master/lib/AuthorizeNetGateway.js#L104-L113). You can compare what is expected to what you get and should find out the problem

**Comment by johntom** [comment]
Hi,
Please note that I'm using  sandbox credentials. 
1) Testing on another computer and the only way to get version 1.0.6 is with npm install -s authorize-net@1.0.6. Leaving out version installs 0.0.5
2) Tesing auth at https://tonicdev.com/npm/authorize-net
```
var AuthorizeNet =require("authorize-net");
var client = new AuthorizeNet({
  API_LOGIN_ID: 'xxxx4Tk9K',
  TRANSACTION_KEY: 'xxxxN24Hqm7C5V98'
});
results
```
API_LOGIN_ID: "xxxx4Tk9K"
TRANSACTION_KEY: "xxxxN24Hqm7C5V98"
endpoint: "https://api.authorize.net/xml/v1/request.api"
```


3) While I know that's the area of the code where it's breaking I'm not sure I how to know what is excpected. 
console shows the following
```
   json.createTransactionResponse.messages[0].message[0].code = [ 'E00007' ]
   json.createTransactionResponse.messages[0].message[0].text = [ 'User     authentication failed due to invalid authentication values.' ] 
   json.createTransactionResponse.transactionResponse = [ '' ] ;
   ```
4) These are the values passed
```
--transactionKey=xxxxN24Hqm7C5V98 --apiLoginId=xxxx4Tk9K --amount=5 --currency=usa --cardnumber=5424000000000015 --cardexpmonth=12 --cardexpyear=2020 --cvv2=999 --billingFirstName=john --billingLastName=Thomas --billingAddress1="500 5th Ave" --billingCity="New York" --billingState=NY --billingPostalCode=10011 --billingCountry=USA
```
Thanks
John

**Comment by johntom** [comment]
I just found the following
E00007 User authentication failed due to invalid authentication values
This is applicable when using our XML or SOAP APIs. This error indicates only invalid credentials. If you have confirmed that your API login ID and transaction key are accurate, you may need to confirm that you are submitting to the correct URL. If you are using a test account, please make sure to post to the sandbox URL and vice versa.

**Comment by johntom** [comment]
Thanks for your support

**Comment by lorenzofox3** [comment]
Yes sorry for that, we should mention the testMode in the doc

**Comment by johntom** [comment]
Thats's the beautiful thing about node machines It's forces you easily to write a descent doc that's easy to follow. I just published 2.0.0 as I accidently bumped to a major version.  I will be creating more machines for your repo.
checkout the following:
http://node-machine.org/
http://node-machine.org/machinepack-authorizenet
Thanks for this great repo.
John
